app/src/main/java/com/paranoidandroid/journey/wizard/utils/JourneyBuilder.java

```diff
@@ -85,6 +85,12 @@ public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
                                 break;
                             }
                         }
+
+                        // use first part of display name if no city is present
+                        if (destination.getCityName() == null) {
+                            String[] components = destination.getDisplayName().replaceAll("\\s+","").split(",");
```

Comment:
Good point! I think I need to set destination and placesId within the buildJourney() method so that by the time onSuccess() is called, those fields have definitely been assigned. I'll make the update now